helpers/string.py

```diff
@@ -0,0 +1,54 @@
+from enum import Enum
+from typing import List
+
+
+class EscapeEnum(Enum):
+    APPEND = "append"
+    PREPEND = "prepend"
+    REPLACE = "replace"
+
+
+class Replacement:
+    def __init__(self, chars: str, output: str, method: EscapeEnum):
+        self.chars = chars
+        self.output = output
+        self.method = method
+
+
+class StringEscaper:
+    """
+    Class to use to escape strings using format defined
+    through a dict.
+
+    Args:
+        escape_def: list of Replacement that defines how to escape
+        characters
+
+        string is escaped by applying method in each Replacement
+        to each char in Replacement.chars using the char in output
+
+        for example:
+            escape_def = [
+                Replacement(chars="<>", output="\\", method=EscapeEnum.PREPEND),
+                Replacement(chars="pre", output="div", method=EscapeEnum.REPLACE)
+            ]
+            StringEscaper(escape_def)
+
+            replace("<pre></pre>")
```

Comment:
If you apply the `replace` function for every _char_ in this `Replacement` definition wouldn't the output actually be `\<divdivdiv\>\</divdivdiv\>` ?

Reply:
good point, that was a bad example, but this also brings up the point that maybe this should include an option for specifying strings to replace, append, or prepend to